gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -153,12 +127,99 @@ static GoogleCloudStorageGrpcReadChannel open(
     }
   }
 
+  private static GoogleCloudStorageGrpcReadChannel openChannel(StorageStubProvider stubProvider,
+      StorageResourceId resourceId, GoogleCloudStorageReadOptions readOptions,
+      BackOffFactory backOffFactory) throws IOException {
+    StorageBlockingStub stub = stubProvider.newBlockingStub();
+    // TODO(b/135138893): We can avoid this call by adding metadata to a read request.
+    //      That will save about 40ms per read.
+    com.google.google.storage.v1.Object storageObject = getObjectMetadata(resourceId, readOptions,
+        stub);
+
+    Preconditions.checkArgument(storageObject != null, "object metadata cannot be null");
+    long footerOffset = Math
+        .max(0, (storageObject.getSize() - (readOptions.getMinRangeRequestSize() / 2)));
+    long footerSize = Math
+        .min(storageObject.getSize(), (readOptions.getMinRangeRequestSize() / 2));
+    ByteString footerContent = getFooterContent(resourceId, readOptions, stub, footerOffset);
```

Comment:
Should we fetch the footer if the file is so small that the entire file can be read in this footer read?

Reply:
The control is upto the user, by configuring `minRangeRequestSize`, I don't see the value bringing in-consistent behaviour based on objectSize.

We could however allow to skip footer caching altogether based on config ( may be determined via file formats or user-config)